starport/cmd/network_chain_revert_launch.go

```diff
@@ -0,0 +1,59 @@
+package starportcmd
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/tendermint/starport/starport/services/network"
+	"github.com/tendermint/starport/starport/services/network/networkchain"
+)
+
+// NewNetworkChainRevertLaunch creates a new chain revert launch command
+// to revert a launched chain.
+func NewNetworkChainRevertLaunch() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "revert-launch [launch-id]",
+		Short: "Revert launch a network as a coordinator",
+		Args:  cobra.ExactArgs(1),
+		RunE:  networkChainRevertLaunchHandler,
+	}
+
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+
+	return c
+}
+
+func networkChainRevertLaunchHandler(cmd *cobra.Command, args []string) error {
+	nb, err := newNetworkBuilder(cmd)
+	if err != nil {
+		return err
+	}
+	defer nb.Cleanup()
+
+	// parse launch ID
+	launchID, err := network.ParseID(args[0])
+	if err != nil {
+		return err
+	}
+
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+
+	chainLaunch, err := n.ChainLaunch(cmd.Context(), launchID)
+	if err != nil {
+		return err
+	}
+
+	if err := n.RevertLaunch(cmd.Context(), launchID); err != nil {
```

Comment:
Maybe RevertLaunch should call nb.Chain internally and ResetGenesisTime? This way we could eliminate having this business logic in cmd package.